akka-cluster/src/main/scala/akka/cluster/MembershipState.scala

```diff
@@ -65,8 +65,16 @@ import akka.annotation.InternalApi
    *         nodes outside of the data center
    */
   lazy val dcReachability: Reachability =
-    overview.reachability.removeObservers(
-      members.collect { case m if m.dataCenter != selfDc ⇒ m.uniqueAddress })
+    overview.reachability.removeObservers(members.collect { case m if m.dataCenter != selfDc ⇒ m.uniqueAddress })
+
+  /**
+   * @return Reachability excluding observations from nodes outside of the data center and observations within self data center,
+   *        but including observed unreachable nodes outside of the data center
+   */
+  lazy val dcReachabilityWithoutObservationsWithin: Reachability =
+    dcReachability.filterRecords { r ⇒
+      latestGossip.member(r.observer).dataCenter != selfDc || latestGossip.member(r.subject).dataCenter != selfDc
```

Comment:
I think that the first part is not needed `latestGossip.member(r.observer).dataCenter != selfDc`
because those observers were already removed by `dcReachability`